### PR TITLE
Add voting to events. Add pagination to eventList. 

### PR DESCRIPTION
**Pull request by iryn34**
## JIRA

* [Main JIRA ticket](https://ssu-jira.softserveinc.com/secure/RapidBoard.jspa?rapidView=777)

## Code reviewers

- [ ] @ashatna
- [ ] @ChristinaPastushchak
- [ ] @dotiurin
- [ ] @iryn34
- [ ] @tarassito
- [ ] @VasilHamulyak
- [ ] @VitalyiKhlivnyi
- [ ] @YurijH

### Second Level Review

- [ ] @ushevtsiv
- [ ] @lhalam
- [ ] @ssOleg

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: eventually/static/src/containers/event/EventList.js [hunk 1]
```diff
@@ -19,23 +22,42 @@ const FloatingButtonStyle = {
...
+const LIMIT = 2;
+
 class EventList extends React.Component {
 
     constructor(props) {
         super(props);
         this.state = {
-            events: []
+            events: [],
+            full_length: 0,
```
Comment (VitalyiKhlivnyi) [inline eventually/static/src/containers/event/EventList.js]: camelCase

File: eventually/static/src/containers/event/EventList.js [hunk 2]
```diff
@@ -64,6 +86,22 @@ class EventList extends React.Component {
             <div style={containerStyle}>
                 {this.getEventLinks()}
                 <CreateEvent />
+                <div
+                    style = { {
```
Comment (VitalyiKhlivnyi) [inline eventually/static/src/containers/event/EventList.js]: remove space between curly brackets

File: eventually/static/src/containers/event/EventList.js [hunk 3]
```diff
@@ -64,6 +86,22 @@ class EventList extends React.Component {
...
+                        width: 500,
+                        margin: '0 auto',
+                        textAlign: 'center'
+                    } }>
```
Comment (VitalyiKhlivnyi) [inline eventually/static/src/containers/event/EventList.js]: remove space between curly brackets

File: eventually/static/src/containers/event/EventList.js [hunk 4]
```diff
@@ -64,6 +86,22 @@ class EventList extends React.Component {
...
+                    <Pagination
+                        total = { this.state.total }
+                        current = { this.state.number }
+                        display = { this.state.display }
+                        onChange = { number => {
+                            this.getData(LIMIT, number + (number - LIMIT), number);
+                            this.setState({ number });
+                        } }/>
```
Comment (VitalyiKhlivnyi) [inline eventually/static/src/containers/event/EventList.js]: remove space between curly brackets

File: eventually/static/src/containers/event/EventList.js (hunk 1 above)
Comment (iryn34) [reply eventually/static/src/containers/event/EventList.js]: Done!

File: eventually/static/src/containers/event/EventList.js (hunk 4 above)
Comment (iryn34) [reply eventually/static/src/containers/event/EventList.js]: Done!

File: eventually/static/src/containers/event/EventList.js (hunk 3 above)
Comment (iryn34) [reply eventually/static/src/containers/event/EventList.js]: Done!

File: eventually/static/src/containers/event/EventList.js (hunk 2 above)
Comment (iryn34) [reply eventually/static/src/containers/event/EventList.js]: Done!

**Review by VitalyiKhlivnyi — DISMISSED** [review]

File: eventually/static/src/containers/event/Event.js [hunk 5]
```diff
@@ -91,6 +90,25 @@ class Event extends React.Component {
         };
     }
 
+    editEvent = (editedEvent) => {
+        console.log(editedEvent);
```
Comment (ashatna) [inline eventually/static/src/containers/event/Event.js]: remove console.log()

File: eventually/static/src/containers/voting/VoteBox.js [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+
+    getData = () => {
+        getVotes(this.state.eventId).then(response => {
+            let sortedVotes = response.data.votes.reverse();
+            let votes = sortedVotes.map(vote => {
+                vote['answers'] = [];
+                return vote;
+            });
+            this.setState({ votes: votes });
+            // this.setState(response.data);
```
Comment (dotiurin) [inline eventually/static/src/containers/voting/VoteBox.js]: delete commented line

File: eventually/static/src/containers/voting/VoteBox.js [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+        this.setState({
+            votes: this.state.votes
+        });
+    };
+
+    addAnswer = (newVote) => {
+        let votes = this.state.votes;
+        for (let i = 0; i < votes.length; i++) {
+            if (votes[i].id == newVote.id) {
+
```
Comment (dotiurin) [inline eventually/static/src/containers/voting/VoteBox.js]: not needed blank line

File: eventually/static/src/containers/event/EventEdit.js [hunk 8]
```diff
@@ -1,6 +1,6 @@
 import React from 'react';
 import { Card, CardActions, CardHeader, CardMedia, CardTitle, CardText } from 'material-ui/Card';
-import { putEventService, GetTeamsListService } from './EventService';
+import { putEventService, GetTeamsListService, getEvent } from './EventService';
```
Comment (ashatna) [inline eventually/static/src/containers/event/EventEdit.js]: getTeamListService - use lowerCamelCase

File: eventually/static/src/containers/event/EventList.js [hunk 9]
```diff
@@ -59,11 +92,66 @@ class EventList extends React.Component {
...
+        const number = 1;
+        this.getData(value, number);
+        this.setState({
+            value: value,
+            limit: value,
+            number: number
+        });
+    };
+
+    // handleChangePagination = number => {
```
Comment (ashatna) [inline eventually/static/src/containers/event/EventList.js]: remove comments

Comment (dotiurin) [inline eventually/static/src/containers/event/EventList.js]: remove commented code

File: eventually/static/src/containers/event/EventService.js [hunk 10]
```diff
@@ -62,7 +71,7 @@ export const GetTeamsListService = () => axios.get(teamPath);
 
 export const PostEventService = (data) => {
     const url = teamPath + data.team + '/event/';
-    axios.post(url, data);
+    return axios.post(url, data);
 };
```
Comment (ashatna) [inline eventually/static/src/containers/event/EventService.js]: postEventService - use lowerCamelCase

File: eventually/static/src/containers/voting/VoteBox.js [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+    getData = () => {
+        getVotes(this.state.eventId).then(response => {
+            let sortedVotes = response.data.votes.reverse();
+            let votes = sortedVotes.map(vote => {
+                vote['answers'] = [];
+                return vote;
+            });
+            this.setState({ votes: votes });
+            // this.setState(response.data);
+        });
```
Comment (ashatna) [inline eventually/static/src/containers/voting/VoteBox.js]: remove comments

**Review by ashatna — DISMISSED** [review]

File: eventually/static/src/containers/event/Event.js (hunk 5 above)
Comment (iryn34) [reply eventually/static/src/containers/event/Event.js]: Done.

File: eventually/static/src/containers/voting/CustomVote.js [hunk 12]
```diff
@@ -0,0 +1,154 @@
...
+import { Link } from 'react-router';
+import { withRouter } from 'react-router-dom';
+import RaisedButton from 'material-ui/RaisedButton';
+import { lightGreen400 } from 'material-ui/styles/colors';
+import {RadioButton, RadioButtonGroup} from 'material-ui/RadioButton';
+import {getUserId} from 'src/helper';
+import ParticipantListDialog from './ParticipantListDialog';
+import {getAnswers} from './VoteService';
+import {putAnswer} from './VoteService';
+
```
Comment (tarassito) [inline eventually/static/src/containers/voting/CustomVote.js:12]: remove blank line

File: eventually/static/src/containers/event/EventList.js (hunk 9 above)
Comment (iryn34) [reply eventually/static/src/containers/event/EventList.js]: Done.

File: eventually/static/src/containers/voting/VoteBox.js (hunk 6 above)
Comment (iryn34) [reply eventually/static/src/containers/voting/VoteBox.js]: Done.

File: eventually/static/src/containers/voting/StandardVote.js [hunk 13]
```diff
@@ -0,0 +1,201 @@
...
+import { lightGreen400 } from 'material-ui/styles/colors';
+import {RadioButton, RadioButtonGroup} from 'material-ui/RadioButton';
+import ActionFavorite from 'material-ui/svg-icons/action/favorite';
+import ActionFavoriteBorder from 'material-ui/svg-icons/action/favorite-border';
+import {getUserId} from 'src/helper';
+import ParticipantListDialog from './ParticipantListDialog';
+import {getParticipants} from './VoteService';
+import {getAnswers} from './VoteService';
+import {putAnswer} from './VoteService';
+
```
Comment (tarassito) [inline eventually/static/src/containers/voting/StandardVote.js:15]: delete blank line

File: eventually/static/src/containers/voting/VoteBox.js (hunk 7 above)
Comment (iryn34) [reply eventually/static/src/containers/voting/VoteBox.js]: Done.

File: eventually/static/src/containers/voting/VoteBox.js (hunk 6 above)
Comment (tarassito) [inline eventually/static/src/containers/voting/VoteBox.js]: delete comment line

File: eventually/authentication/views.py [hunk 14]
```diff
@@ -262,4 +262,21 @@ def get_all_users(request):
         data = {'users': [user.to_dict() for user in users]}
         return JsonResponse(data, status=200)
     return RESPONSE_400_INVALID_HTTP_METHOD
-    
\ No newline at end of file
+
+def get_selected_users(request):
```
Comment (lhalam) [inline eventually/authentication/views.py]: to improve

**Review by lhalam — CHANGES_REQUESTED** [review]

File: eventually/utils/validators.py [hunk 15]
```diff
@@ -267,6 +268,31 @@ def event_data_validate(data, required_keys):
     is_data_valid = len(errors) == 0
     return is_data_valid
 
+def event_paginator_validate(request):
```
Comment (lhalam) [inline eventually/utils/validators.py]: This method should only validate the data! Does not have to return a new dictionary, it should return True or False.

File: eventually/vote/models.py [hunk 16]
```diff
@@ -241,7 +241,7 @@ def update(self, members=None, text=None):
 
         :return: none
         """
-        if members:
+        if members or members == []:
```
Comment (lhalam) [inline eventually/vote/models.py]: use `members  is not None`:

File: eventually/vote/views.py [hunk 17]
```diff
@@ -353,3 +367,27 @@ def delete(self, request, team_id, event_id, vote_id, answer_id=None):  # pylint
 
         Answer.delete_by_id(answer_id)
         return RESPONSE_200_DELETED
+
+def extended_answers(request, event_id, vote_id):
```
Comment (lhalam) [inline eventually/vote/views.py]: Rename this method.  for example `get_answers_with_member`.
And make it a class AnswerView method.

**Review by lhalam — COMMENTED** [review]
to improve

File: eventually/static/src/containers/event/EventEdit.js (hunk 8 above)
Comment (iryn34) [reply eventually/static/src/containers/event/EventEdit.js]: Done.

File: eventually/static/src/containers/event/EventService.js (hunk 10 above)
Comment (iryn34) [reply eventually/static/src/containers/event/EventService.js]: Done.

File: eventually/static/src/containers/voting/CustomVote.js (hunk 12 above)
Comment (iryn34) [reply eventually/static/src/containers/voting/CustomVote.js:12]: Done.

File: eventually/static/src/containers/voting/StandardVote.js (hunk 13 above)
Comment (iryn34) [reply eventually/static/src/containers/voting/StandardVote.js:15]: Done.

File: eventually/utils/validators.py (hunk 15 above)
Comment (iryn34) [reply eventually/utils/validators.py]: Done.

File: eventually/vote/models.py (hunk 16 above)
Comment (iryn34) [reply eventually/vote/models.py]: Done.

**Review by lhalam — APPROVED** [review]

**Pull request merged**